src/main/java/com/example/tradingapp/trading/OrderRequestEncoderImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.example.tradingapp.trading;
+
+import com.example.tradingapp.secrets.Secrets;
+import com.example.tradingapp.trading.model.OkxOrderRequest;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.codec.digest.HmacAlgorithms;
+import org.apache.commons.codec.digest.HmacUtils;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.StringEntity;
+import org.springframework.stereotype.Component;
+
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+
+@Component
+public class OrderRequestEncoderImpl implements OrderRequestEncoder {
+
+    private final String timestamp = String.valueOf(Instant.now().truncatedTo(ChronoUnit.MILLIS));
+    private final String PATH = "/api/v5/trade/order";
+
+    private final String API_KEY = Secrets.API_KEY;
+    private final String SECRET_KEY = Secrets.SECRET_KEY;
+    private final String PASSPRHASE = Secrets.PASSPHRASE;
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public HttpRequestBase encode(OkxOrderRequest orderRequest) throws JsonProcessingException, UnsupportedEncodingException {
```

Comment:
We will use `OrderRequest` type in this class, to incapsulate it from generic logic.
This way, we can create `OkxOrderRequest` in this method's body based on `OrderRequest` parameter with simple mapping